### ISSUE TITLE
new Apple TV4K and remote compatibility issues

### ISSUE THREAD

**Issue by superandrew** [yoshisuga/MAME4iOS#332]
Hi, I just built the latest version from git and I noticed that there are issues with the new Apple TV 4k or I should say probably with the new remote.

1. When inside a game I am not able to fire the menu with the remote. 
2. Even when I use a bt controller to fire a remote and click x as I usually did to exit to Mame, it stays inside the game. The only way to go to Mame4ios is to kill it.

Since I imagine not much people already has a new apple tv, if you help me find the point where to look I might be able to take a look at it in debug.

**Comment by ToddLa** [comment]
I get a few remotes today, expect a fix tomorrow!!

**Comment by ToddLa** [comment]
The old MENU button is now labeled BACK on the new remote. 

But I assume it still sends the same button events, sooo BACK should bring up the menu.

**Comment by superandrew** [comment]
I thought the same, but it doesn't seem to be working

**Comment by ToddLa** [comment]
> Even when I use a bt controller to fire a remote and click x as I usually did to exit to Mame, it stays inside the game. The only way to go to Mame4ios is to kill it.

This is related to something else, not the new remote.
I will verify that as well.

**Comment by superandrew** [comment]
correct, but to be honest I can't figure out what it is unless it depends on something introduced recently because I updated from git and did a new build

**Comment by ToddLa** [comment]
I just tried the new remote, works fine (as I expected MENU and BACK are same button)

I am running PR #327 (with MAME 139).

I will go give `master` a quick try.

**Comment by ToddLa** [comment]
Yup `master` (and 139) works fine....

What exact problem are you seeing?

I am on a TV gen 4 with new remote, maybe your problems needs the new TV gen 6???

**Comment by ToddLa** [comment]
If it is easy try my PR #327, it has some fixes for the TV. (But nothing to change how the remote works...)

Any are you using MAME 231? Or 139?

**Comment by ToddLa** [comment]
> Even when I use a bt controller to fire a remote and click x as I usually did to exit to Mame, it stays inside the game. The only way to go to Mame4ios is to kill it.

Yes please run PR #327, this is one of the things I fixed (after I broke it....) the exit procedure is slightly changes to support newMAME.

**Comment by ToddLa** [comment]
Yup new remote works just fine on Apple TV Gen 4, but on brand new Gen 6? (Or 4K Gen 2) TV it does not work!!

Debugging now.

**Comment by ToddLa** [comment]
Yup I see the problem, deprecated apis stop working, who knew!

Strange that it only happens on the Gen 6, I am running latest tvOS on my gen 4, and new remote works great.

I know how to fix, but we were keeping the old code around for MFi devices, so I will need to test on my MFi and Xbox and 8BitDo, etc...

**Comment by superandrew** [comment]
I have a steelseries and of course the new remote. Let me know if I can help debugging in some way

**Comment by ToddLa** [comment]
Strange only fails when connected to a 4K HDR display!

On my test 1080p LCD, remote works!

**Comment by ToddLa** [comment]
Thanks @superandrew i will let you know when I do a push, I just don't want to break MFi controllers, cuz the code I need to change is in the MFi path...

**Comment by ToddLa** [comment]
@superandrew I did I push to my PR, minimal change to not use the old skool pausedHandler on tvOS 14....

**Comment by ToddLa** [comment]
Strange I re-connect to my 4K, and now the Nimbus does not work, and I don't just mean in M4i the Nimbus does not work system wide at all, it connects just fine but does not work to drive the tvOS UX....???

Time to reboot tvOS...

**Comment by superandrew** [comment]
I thought it was me... I have to reconnect the nimbus from scratch everytime with this Apple TV otherwise it connects but it doesn't work

**Comment by ToddLa** [comment]
PR #327 has a fix for this

**Comment by superandrew** [comment]
@ToddLa I thought it was at system level, the nimbus seemed not to work even with standard ATV menus

**Comment by ToddLa** [comment]
The change was needed to fix the apple remote, but also effects the Nimbus path, so I wanted to test with a Numbus that is why this issue talks about Nimbus.

**Comment by ToddLa** [comment]
This was fixed in #327
